src/platform/tls_schannel.c

```diff
@@ -2298,8 +2298,11 @@ CxPlatTlsWriteDataToSchannel(
                         TlsContext->Connection,
 #ifdef _KERNEL_MODE
                         (void*)&PeerCert,
+                        NULL,
 #else
-                        (void*)PeerCert,
+                        (QUIC_CERTIFICATE*)PeerCert,
+                        (QUIC_CERTIFICATE_CHAIN*)(PeerCert ? PeerCert->hCertStore : NULL),
```

Comment:
I _beleive_ `PeerCert` is guaranteed to be non-null here. Though, we should put a `CXPLAT_DBG_ASSERT` above for this.

Reply:
What value would it have on server if client did not provide certificate? 

For SslStream, the sever can request client certificate and when handshake is done we can call validation callback with no certificate. That gives application control to decide if that is ok or not. e.g. use the certificate opportunistically.

Reply:
I guess I'd expect `QueryContextAttributesW` to fail.

Reply:
I can run more tests as well I'm fine with adding the assert as you suggested.

Reply:
updated.